test/utils/EnumerableSet.test.js

```diff
@@ -0,0 +1,54 @@
+const { contract } = require('@openzeppelin/test-environment');
+
+const { expect } = require('chai');
+
+const EnumerableSetMock = contract.fromArtifact('EnumerableSetMock');
+
+const a = '0x0000000000000000000000000000000000000001';
+const b = '0x0000000000000000000000000000000000000002';
+const c = '0x0000000000000000000000000000000000000003';
+
+/** @test {EnumerableSet} contract */
+describe('EnumerableSet', function () {
+  beforeEach(async function () {
+    this.set = await EnumerableSetMock.new();
+  });
+
+  it('contains can return false.', async function () {
+    expect(await this.set.testContains(a)).to.equal(false);
+  });
+
+  it('adds a value.', async function () {
+    await this.set.testAdd(a);
+    expect(await this.set.testContains(a)).to.equal(true);
+  });
+
+  it('adds several values.', async function () {
```

Comment:
Can you add tests for situations where the same value is added multiple times?

Reply:
This wasn't caught by coverage because there is no `else` branch (same below with `remove`).